R/pretty_lollipop_plot.R

```diff
@@ -2,58 +2,118 @@
 #'
-#' @param maf_df A data frame containing the mutation data (from a MAF).
-#' @param gene The gene symbol to plot.
-#' @param plot_title Optional (defaults to gene name).
-#' @param plot_theme Options: cbioportal(default), blue, simple, nature, nature2, ggplot2, and dark.
-#' @param out_name Optional, set the file name of the plot, if you export it to disk. Default name is my_lollipop_plot_{gene}.
-#'
-#' @return Nothing.
+#' @param maf_data A data frame containing the mutation data (from a MAF).
+#' @param gene_symbol The gene symbol to plot.
+#' 
+#' @return A lolipop plot.
 #'
-#' @import g3viz dplyr
+#' @import tidyverse BiocManager maftools
 #' @export
 #'
 #' @examples
 #' library(GAMBLR.data)
 #' 
-#' #get metadata (Fl and DLBCL)
-#' metadata = get_gambl_metadata()
-#' this_meta = dplyr::filter(metadata, pairing_status == "matched")
-#' this_meta = dplyr::filter(this_meta, pathology %in% c("FL", "DLBCL"))
+#' #get metadata (BL_Thomas)
+#' metadata <- get_gambl_metadata() %>%
+#'     filter(cohort == "BL_Thomas")
 #' 
-#' maf = get_ssm_by_sample(these_samples_metadata = this_meta)
+#' maf_data <- get_ssm_by_samples(
+#'     these_samples_metadata = metadata
+#' )
 #'
 #' #construct pretty_lollipop_plot.
-#' pretty_lollipop_plot(maf_df = maf,
-#'                      gene = "MYC",
-#'                      plot_title = "Mutation data for MYC",
-#'                      plot_theme = "nature2")
+#' lolipop_result <- get_lolipop_by_gene_symbol(maf_data, "MYC")
+#' print(lolipop_result)
 #'
-pretty_lollipop_plot = function(maf_df,
-                                gene,
-                                plot_title,
-                                plot_theme = "cbioportal",
-                                out_name = paste0("my_lollipop_plot_", gene)){
-  if(missing(gene)){
-    stop("Plese provide a gene...")
+get_lolipop_by_gene_symbol <- function(maf_data, 
+                                       gene_symbol = NULL) {
+  # Read the MAF object if it's not already in MAF format
+  if (!inherits(maf_data, "MAF")) {
+    maf_data <- read.maf(maf_data)
   }
+  
+  maf_data <- maf_data@data 
+  # Filter for the specific gene 
+  gene_df <- maf_data %>% 
+    filter(Hugo_Symbol == gene_symbol) %>%
+    mutate(AA = as.numeric(gsub("[^0-9]+", "", gsub("([0-9]+).*", "\\1", HGVSp_Short)))) %>%
+    arrange(AA)
 
-  if(missing(plot_title)){
-    plot_title = gene
+  gene_counts <- gene_df %>% 
+    group_by(AA, Start_Position, 
+             End_Position, 
+             Variant_Classification, 
+             Reference_Allele, 
+             Tumor_Seq_Allele2) %>%
+    arrange(AA) %>%
+    summarise(mutation_count = n()) 
+
+  protein_domain_subset <- subset(protein_domains, HGNC == gene_symbol)
+  if (!is.data.frame(protein_domain_subset)) {
+    protein_domain_subset <- as.data.frame(protein_domain_subset)
   }
+   
+  domain_data <- protein_domain_subset %>% data.frame(start.points = protein_domain_subset$Start,
```

Comment:
We can use this as opportunity to stick to a proper formatting. [Here](https://github.com/morinlab/GAMBLR.data/blob/e6592af5697bcc090691c8d50d413108a48aa733/R/get_ssm_by_regions.R#L126-L144) might be a good reference eexample. Notice how each pipe ends the line and the next one starts with indentation. This way it is easier to read the code top-to-bottom instead of left-to-right